components/camel-opensearch/src/test/java/org/apache/camel/component/opensearch/integration/OpensearchGetSearchDeleteExistsUpdateIT.java

```diff
@@ -0,0 +1,913 @@
+        actualQuery.put("doc.testSearchWithMapQuery1", "bar");
+        Map<String, Object> match = new HashMap<>();
+        match.put("match", actualQuery);
+        Map<String, Object> query = new HashMap<>();
+        query.put("query", match);
+        HitsMetadata<?> response = template.requestBody("direct:search", query, HitsMetadata.class);
+        assertNotNull(response, "response should not be null");
+        assertNotNull(response.total());
+        assertEquals(0, response.total().value(), "response hits should be == 0");
+
+        // Match
+        actualQuery.put("doc.testSearchWithMapQuery1", "foo");
+        // the result may see stale data so use Awaitility
+        Awaitility.await().atMost(10, TimeUnit.SECONDS).untilAsserted(() -> {
+            HitsMetadata<?> resp = template.requestBody("direct:search", query, HitsMetadata.class);
+            assertNotNull(resp, "response should not be null");
+            assertNotNull(resp.total());
+            assertEquals(1, resp.total().value(), "response hits should be == 1");
+            assertEquals(1, resp.hits().size(), "response hits should be == 1");
+            Object result = resp.hits().get(0).source();
+            assertInstanceOf(ObjectNode.class, result);
+            assertTrue(((ObjectNode) result).has("doc"));
+            JsonNode node = ((ObjectNode) result).get("doc");
+            assertTrue(node.has("testSearchWithMapQuery1"));
+            assertEquals("foo", node.get("testSearchWithMapQuery1").asText());
+        });
+    }
+
+    @Test
+    void testSearchWithStringQuery() {
+        //first, Index a value
+        Map<String, String> map1 = Map.of("testSearchWithStringQuery1", "foo");
+        Map<String, String> map2 = Map.of("testSearchWithStringQuery2", "bar");
+        Map<String, Object> headers = new HashMap<>();
+        headers.put(OpensearchConstants.PARAM_OPERATION, OpensearchOperation.Bulk);
+        headers.put(OpensearchConstants.PARAM_INDEX_NAME, "twitter");
+        template.requestBodyAndHeaders("direct:start", List.of(Map.of("doc", map1), Map.of("doc", map2)), headers,
+                String.class);
+
+        // No match
+        String query = """
+                {
+                    "query" : { "match" : { "doc.testSearchWithStringQuery1" : "bar" }}
+                }
+                """;
+
+        HitsMetadata<?> response = template.requestBody("direct:search", query, HitsMetadata.class);
+        assertNotNull(response, "response should not be null");
+        assertNotNull(response.total());
+        assertEquals(0, response.total().value(), "response hits should be == 0");
+
+        // Match
+        String q = """
+                {
+                    "query" : { "match" : { "doc.testSearchWithStringQuery1" : "foo" }}
+                }
+                """;
+        // the result may see stale data so use Awaitility
+        Awaitility.await().atMost(10, TimeUnit.SECONDS).untilAsserted(() -> {
```

Comment:
I think it would be better to split this so that `untilAsserted` tests for only 1 assertion. Combining them usually makes it harder to identify the root cause of failures.

The same note for all the others using the same pattern.